src/ReaderScreen/components/AudioPlayer/hooks/useTrackPlayer.js

```diff
@@ -0,0 +1,160 @@
+import { useState, useEffect } from "react";
+import TrackPlayer, {
+  usePlaybackState,
+  useProgress,
+  State,
+  AppKilledPlaybackBehavior,
+} from "react-native-track-player";
+import { useSelector } from "react-redux";
+import { addTrack, playTrack, pauseTrack, stopTrack, resetPlayer } from "@common/TrackPlayerUtils";
+import { logError, logMessage } from "@common";
+
+const useTrackPlayer = () => {
+  const [isInitialized, setIsInitialized] = useState(false);
+  const playbackState = usePlaybackState();
+  const progress = useProgress();
+  const [isPlaying, setIsPlaying] = useState(false);
+  const isAudio = useSelector((state) => state.isAudio);
+
+  useEffect(() => {
+    const setupPlayer = async () => {
+      try {
+        // Check if player is already initialized
+        const isSetup = await TrackPlayer.isServiceRunning();
+        if (!isSetup) {
+          // Initialize player if not already set up
+          await TrackPlayer.setupPlayer({
+            waitForBuffer: true,
+            maxCacheSize: 1024,
+            iosCategory: "playback",
+            alwaysPauseOnInterruption: true,
+          });
+          await TrackPlayer.updateOptions({
+            android: {
+              appKilledPlaybackBehavior:
+                AppKilledPlaybackBehavior.StopPlaybackAndRemoveNotification,
+            },
+            capabilities: [
+              TrackPlayer.Capability.Play,
+              TrackPlayer.Capability.Pause,
+              TrackPlayer.Capability.SkipToNext,
+              TrackPlayer.Capability.SkipToPrevious,
+              TrackPlayer.Capability.Stop,
+              TrackPlayer.Capability.SeekTo,
+            ],
+            compactCapabilities: [
+              TrackPlayer.Capability.Play,
+              TrackPlayer.Capability.Pause,
+              TrackPlayer.Capability.SkipToNext,
+            ],
+          });
+        }
+        setIsInitialized(true);
+      } catch (error) {
+        logError("Error initializing TrackPlayer:", error);
+      }
+    };
+
+    setupPlayer();
+  }, []);
```

Comment:
How about adding a cleanup function to properly dispose of the TrackPlayer when the component unmounts. Also, the setup could be moved to a singleton service to avoid multiple initializations

Reply:
done